xbmc/music/MusicDatabase.cpp

```diff
@@ -7489,9 +7489,10 @@ unsigned int CMusicDatabase::GetSongIDs(const Filter &filter, std::vector<std::p
     if (NULL == m_pDB.get()) return 0;
     if (NULL == m_pDS.get()) return 0;
 
-    std::string strSQL = "select idSong from songview ";
+    std::string strSQL = "SELECT idSong FROM songview ";
     if (!CDatabase::BuildSQL(strSQL, filter, strSQL))
       return false;
+    strSQL += PrepareSQL(" ORDER BY RANDOM()");
```

Comment:
This looks confusing as at first it doesn't seem to make sense it "orders by random" by default, but apparently GetSongIDs is only used by PartyMode. Perhaps add a big fat comment why or make random optional by adding an additional argument to GetSongIDS() to make it explicit?

Reply:
`CMusicDatabase::GetSongIDs` fetches a vector of <1, ID> pairs. This result structure and method is only used by party mode and always wants to be randomly  sorted. Starting to document what calls what in code comments, something that can easily be searched and just as easy get out of date,  does not seem a good idea to me.

Reply:
I concur with @arnova. The method name does not make it obvious at all that you will get randomly sorted results, making it quite confusing. You should either change the method name or add a parameter.

Reply:
Shall I call it `GetRandomSongIdVectorForPartyMode` ?

Reply:
I think "GetRandomSongIDs" should be fine ;-)

Reply:
OK, I have renamed the partymode GetXXXIDs methods in both db and added Doxygen comments. 
I also got so carried away I even removed some old unused methods too that were part of party mode years ago and got left behind. Now I want my cookie ;-)

Reply:
:cookie:

Reply:
> OK, I have renamed the partymode GetXXXIDs methods in both db and added Doxygen comments.
> I also got so carried away I even removed some old unused methods too that were part of party mode years ago and got left behind. Now I want my cookie ;-)

Thanks for the effort. Like I said: this part of the code didn't get any love for a very long time. I'm fine with how it is now so I think we can push this for 18.2?

Btw. have another cookie ;-)